object-manipulation-2/src/pick.js

```diff
@@ -0,0 +1,21 @@
+/* exported pick */
+function pick(source, keys) {
+  var newObj = {};
+  for (var i in source) {
+    for (var j = 0; j < keys.length; j++) {
+      if (keys[j] === i) {
+        if (typeof source[i] !== 'undefined') {
+          newObj[i] = source[i];
+        }
+      }
```

Comment:
You can remove the exterior `for...in` loop by adjusting this to `if(source[keys[i]] !== undefined)`